contracts/SponsorFaucet.sol

```diff
@@ -180,21 +214,30 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard, WhitelistAdminRole
         );
     }
 
+    /*** internal helper functions ***/
+
     function _isAppliableForGas(address dapp) internal returns (bool) {
-        if(!dapp.isContract()) return false;
+        if (!dapp.isContract()) return false;
         uint256 gas_balance = internal_sponsor.getSponsoredBalanceForGas(dapp);
         address current_sponsor = internal_sponsor.getSponsorForGas(dapp);
-        uint256 contract_gas_bound = dapp_bounds[small].gas_bound;
-        uint256 contract_gas_total_limit = dapp_bounds[small].gas_total_limit;
+        uint256 contract_gas_bound;
```

Comment:
There're still many duplicated codes, how about to define a method as below:
```
function _getBounds(address dapp) internal returns (bounds) {
    if (isCustome(dapp)) return custom_bounds;
    if (isLargeContract(dapp)) return large_bounds;
    return small_bounds;
}
```

Reply:
**_getBounds** added, duplicated codes removed